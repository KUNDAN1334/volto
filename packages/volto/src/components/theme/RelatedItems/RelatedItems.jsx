/**
 * RelatedItems component.
 * @module components/theme/RelatedItems/RelatedItems
 */

import React from 'react';
import { UniversalLink } from '@plone/volto/components';
import { defineMessages, injectIntl } from 'react-intl';
import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';

const messages = defineMessages({
  relatedItems: {
    id: 'Related Items',
    defaultMessage: 'Related Items',
  },
});

/**
 * Related Items component.
 * @function RelatedItems
 * @param {array} tags Array of related items.
 * @returns {string} Markup of the component.
 */

const RelatedItems = ({ relatedItems, intl }) =>
  relatedItems && relatedItems.length > 0 ? (
    <Container>
      <h4 className="ui header">{intl.formatMessage(messages.relatedItems)}</h4>
      <div className="ui list">
        {relatedItems.map((relatedItem) =>
          relatedItem ? (
            <div className="item" key={relatedItem['@id']}>
              <div className="content">
                <UniversalLink className="header" item={relatedItem}>
                  <div>{relatedItem['title']}</div>
                  <div className="description">
                    {relatedItem['description']}
                  </div>
                </UniversalLink>
              </div>
            </div>
          ) : null,
        )}
      </div>
    </Container>
  ) : (
    <span />
  );

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
RelatedItems.propTypes = {
  relatedItems: PropTypes.arrayOf(
    PropTypes.shape({
      '@id': PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
    }),
  ),
};

/**
 * Default properties.
 * @property {Object} defaultProps Default properties.
 * @static
 */
RelatedItems.defaultProps = {
  RelatedItems: null,
};

export default injectIntl(RelatedItems);
