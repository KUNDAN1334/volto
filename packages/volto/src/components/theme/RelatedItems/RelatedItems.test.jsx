import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { MemoryRouter } from 'react-router-dom';

import RelatedItems from './RelatedItems';

const mockStore = configureStore();

describe('Related Items', () => {
  it('renders without related items', () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <RelatedItems />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders with related items', () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const relatedItems = [
      {
        '@id': '/test-1',
        title: 'Title 1',
        description: 'Description 1',
      },
      {
        '@id': '/test-2',
        title: 'Title 2',
        description: 'Description 2',
      },
    ];
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <RelatedItems relatedItems={relatedItems} />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders with related items has null', () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const relatedItems = [
      {
        '@id': '/test-1',
        title: 'Title 1',
        description: 'Description 1',
      },
      null,
    ];
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <RelatedItems relatedItems={relatedItems} />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
