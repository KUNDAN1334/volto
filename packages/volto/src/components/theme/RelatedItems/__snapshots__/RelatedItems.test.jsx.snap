// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Related Items renders with related items 1`] = `
<div
  className="ui container"
>
  <h4
    className="ui header"
  >
    Related Items
  </h4>
  <div
    className="ui list"
  >
    <div
      className="item"
    >
      <div
        className="content"
      >
        <a
          className="header"
          href="/test-1"
          onClick={[Function]}
          target={null}
          title={null}
        >
          <div>
            Title 1
          </div>
          <div
            className="description"
          >
            Description 1
          </div>
        </a>
      </div>
    </div>
    <div
      className="item"
    >
      <div
        className="content"
      >
        <a
          className="header"
          href="/test-2"
          onClick={[Function]}
          target={null}
          title={null}
        >
          <div>
            Title 2
          </div>
          <div
            className="description"
          >
            Description 2
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Related Items renders with related items has null 1`] = `
<div
  className="ui container"
>
  <h4
    className="ui header"
  >
    Related Items
  </h4>
  <div
    className="ui list"
  >
    <div
      className="item"
    >
      <div
        className="content"
      >
        <a
          className="header"
          href="/test-1"
          onClick={[Function]}
          target={null}
          title={null}
        >
          <div>
            Title 1
          </div>
          <div
            className="description"
          >
            Description 1
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Related Items renders without related items 1`] = `<span />`;
